SYCL/ESIMD/api/functional/functions/functions_atomic_update_inc_dec.cpp

```diff
@@ -0,0 +1,250 @@
+          typename AlgorithmToChangeT>
+class run_test {
+  static constexpr int NumElems = NumElemsT::value;
+  static constexpr int NDRangeDim = NDRangeDimT::value;
+  static constexpr int NumElemsToChange = NumElemsToChangeT::value;
+  static constexpr atomic_op ChosenOperator = ChosenOperatorT::value;
+  static constexpr functions::offset AlgorithmToChange =
+      AlgorithmToChangeT::value;
+  using TestDescriptionT = functions::TestDescription<NumElems, TestCaseT>;
+
+public:
+  bool operator()(sycl::queue &queue, const std::string &data_type,
+                  const std::string &operator_type) {
+    static_assert(NDRangeDim <= 3 && NDRangeDim > 0);
+    static_assert(NumElemsToChange <= NumElems);
+    bool passed = true;
+    constexpr DataT base_value = 100;
+    constexpr DataT result_fill_value = 5;
+
+    auto init_values =
+        functions::get_init_values<NumElemsToChange, ChosenOperator, DataT>(
+            base_value);
+    shared_allocator<DataT> allocator(queue);
+    shared_vector<DataT> shared_init_values(init_values.begin(),
+                                            init_values.end(), allocator);
+    shared_vector<DataT> shared_result(NumElems, result_fill_value, allocator);
+
+    auto offsets =
+        functions::get_offsets<NumElemsToChange, AlgorithmToChange>();
+    shared_vector<size_t> shared_offsets(offsets.begin(), offsets.end(),
+                                         shared_allocator<size_t>(queue));
+
+    constexpr int NumberIteractions = 16;
+    sycl::nd_range<NDRangeDim> nd_range =
+        get_sycl_nd_range<NDRangeDim>(NumberIteractions);
+
+    queue.submit([&](sycl::handler &cgh) {
+      DataT *shared_init_values_ptr = shared_init_values.data();
+      DataT *shared_result_ptr = shared_result.data();
+      size_t *const offsets_ptr = shared_offsets.data();
+
+      cgh.parallel_for<
+          Kernel<DataT, NumElems, TestCaseT, ChosenOperatorT, NDRangeDimT,
+                 NumElemsToChangeT, ChangeElemFilterT, AlgorithmToChangeT>>(
+          nd_range, [=](sycl::nd_item<1> nd_item) SYCL_ESIMD_KERNEL {
+            const size_t work_item_index = nd_item.get_global_linear_id();
+            TestCaseT::template call_esimd_function<
+                DataT, NumElems, NumElemsToChange, ChangeElemFilterT,
+                ChosenOperator>(shared_init_values_ptr, shared_result_ptr,
+                                offsets_ptr, work_item_index);
+          });
+    });
+    queue.wait_and_throw();
+
+    std::sort(shared_result.begin(), shared_result.end());
+
+    ChangeElemFilterT filter{};
+    size_t num_changed_elems = 0;
+    // Collect the number of elements that should be changed. This value will be
+    // used to calculate expected value for all elements that will be returned
+    // after atomic_update call the sum of changed elements.
+    for (size_t i = 0; i < NumElemsToChange; ++i) {
+      if (filter(i) == 1) {
+        ++num_changed_elems;
+      }
```

Comment:
```suggestion
      num_changed_elems += (filter(i) == 1);
```

Reply:
this comment has been applied in this commit: ae69b2dc7d65f231687fb2bdeadb72190d6ef384